src/server/multi_command_squasher.cc

```diff
@@ -163,7 +163,7 @@ bool MultiCommandSquasher::ExecuteStandalone(facade::RedisReplyBuilder* rb, Stor
 
   if (cmd->Cid()->IsTransactional())
     tx->InitByArgs(cntx_->ns, cntx_->conn_state.db_index, args);
-  service_->InvokeCmd(cmd->Cid(), args, rb, cntx_);
+  service_->InvokeCmd(cmd->Cid(), args, rb, cntx_, cmd->Cid()->name());
```

Comment:
If the solution was to pass `cmd->Cid()->name()` everywhere we could actually simplify this API a lot. Unfortunately, this is not true. 

As I explained in the previous PR, we break `multi/exec` and `squashing`. This can be easily verified with a unit test:

```
multi
__set foo bar
exec

info all
``` 

And you will see `cmdstat_set:calls=1` instead of `cmdstat___set:calls=1` which is a bug! 

We still pass the original command name unconditionally which does not cover all cases. The same applies for squashing

Reply:
My bad, I missed this, I think we discussed before that we need extra context in `StoredCmd` to keep around the original command name. I will add this to the PR.

Reply:
Folks, this is a minor feature requested internally and if it unwinds to complicated - really we can just kill it. I do not want to complicate the codebase for this.

Reply:
Well, @abhijat if it turns out to be impractical as @romange suggests just leave it as is. In fact, we can simplify this a lot if we "allow some room for incorrectly measuring stats"

Reply:
as long as @romange and @adiholden are ok with this I am happy :smile: I leave the final approve to them.

Reply:
It's the first time I am looking at this PR and the one before. 
I believe the whole implementation can be simplified by  removing `cmd_aliases_` altogether.
Instead, aliases should be added to the main `cmd_map_` by duplicating (deep copying) the CommandId objects that they point to. Since that point Dragonfly code does not need to know about a special nature of aliases so the feature will be local to `CommandRegistry::CommandRegistry()`. Please let me know if you think it won't work.

Reply:
I think the test case that Kostas suggested should be part of the PR in case we go along with the (simple) implementation.

Reply:
Looking at the code it seems like having duplicate command ids would be fine, the only way it might be a problem is if in future if we ever expect command ids in the map to be unique for some reason. 

It looks like right now we only iterate over commands in the map in one place to initialize the stats (in `CommandRegistry::Init`), which should be fine because the stats are local to the `CommandId` object and not some global state.

Reply:
command ids are unique, they have differrent names too.

Regarding the iteration - you are right - we send all the commands during the "COMMAND" command and hiding aliases indeed makes sense. For that you can use the HIDDEN in CommandOpt in command_registry.h.

Reply:
see `Service::Command` implementation

Reply:
ok, I will create a new pr with the clone approach.